src/public_key.rs

```diff
@@ -19,15 +19,19 @@ use std::{
 };
 use threshold_crypto;
 
+/// Wrapper for different public key types.
 #[derive(Clone, Copy, Eq, PartialEq, Serialize, Deserialize)]
-#[allow(clippy::large_enum_variant)]
 pub enum PublicKey {
+    /// Ed25519 public key.
     Ed25519(ed25519_dalek::PublicKey),
+    /// BLS public key.
     Bls(threshold_crypto::PublicKey),
+    /// BLS public key share.
     BlsShare(threshold_crypto::PublicKeyShare),
 }
 
 impl PublicKey {
+    /// Returns `true` if `signature` matches the message.
```

Comment:
This function does not return a boolean.
Should be something like
```rust
/// Returns `Ok(())` if `signature` matches the message and `Err(Error::InvalidSignature)` otherwise.
```